classes/4.html

```diff
@@ -56,16 +56,365 @@ <h3>¿Qué es React?</h3>
+                  super(props);
+                  this.state = {fecha: new Date()};
+                }
+              
+                render() {
+                  return (
+                    <div>
+                      <h1>Hola, mundo!</h1>
+                    </div>
+                  );
+                }
+              }
+            </code></pre>
+          </section>
+          <section>
+            <p>
+              Hemos definido el estado pero no lo hemos usado aún. Vamos a mostrar en pantalla el estado:
+            </p>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              class Reloj extends React.Component {
+                constructor(props) {
+                  super(props);
+                  this.state = {fecha: new Date()};
+                }
+              
+                render() {
+                  return (
+                    <div>
+                      <h1>Hola, mundo!</h1>
+                      <h2>Son las {this.state.fecha.toLocaleTimeString()}.</h2>
+                    </div>
+                  );
+                }
+              }
+            </code></pre>
+          </section>
+          <section>
+            <p>
+              Como se pudo ver, se hace de la misma forma que para las props, pero usando <b>state</b>.
+            </p>
+          </section>
+          <section>
+            <p>
+              Queremos configurar un temporizador cada vez que <b>Reloj</b> se renderice en el DOM por primera vez. Esto se llama «montaje» en React.
+              También queremos borrar ese temporizador cada vez que el DOM producido por <b>Reloj</b> se elimine. Esto se llama «desmontaje» en React.
+              Podemos declarar métodos especiales en la clase del componente para ejecutar algún código cuando un componente se monta y desmonta:
+            </p>
+          </section>
+          <section>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              class Reloj extends React.Component {
+                constructor(props) {
+                  super(props);
+                  this.state = {fecha: new Date()};
+                }
+
+                componentDidMount() {
+                }
+              
+                componentWillUnmount() {
+                }
+              
+                render() {
+                  return (
+                    <div>
+                      <h1>Hola, mundo!</h1>
+                      <h2>Son las {this.state.fecha.toLocaleTimeString()}.</h2>
+                    </div>
+                  );
+                }
+              }
+            </code></pre>
+            <p>
+              Estos métodos son llamados «métodos de ciclo de vida» del componente (lifecycle methods).
+            </p>
+          </section>
+          <section>
+            <p>
+              El método componentDidMount() se ejecuta después que la salida del componente ha sido renderizada en el DOM.
```

Comment:
```suggestion
              El método <b>componentDidMount()</b> se ejecuta después que la salida del componente ha sido renderizada en el DOM.
```

yo lo pondria en negrita o destacado de alguna forma, pero es sugerencia nomás 😄